exporters/holstebro_helpers.py

```diff
@@ -0,0 +1,811 @@
+
+SAML_TOKEN = os.environ.get('SAML_TOKEN', None)
+
+logger = logging.getLogger('holstebro-helpers')
+
+
+cfg_file = pathlib.Path.cwd() / 'settings' / 'holstebro.settings.json'
+if not cfg_file.is_file():
+    raise Exception('No setting file')
+SETTINGS = json.loads(cfg_file.read_text(encoding='utf-8'))
+
+
+def holstebro_generic_export(mh, nodes, filename):
+
+    fieldnames = ['uuid',
+                  'name',
+                  'email',
+                  'sam_account',
+                  'phone',
+                  'location',
+                  'mobile',
+                  'loen_nummer',
+                  'forvaltning',  # direktørområde
+                  'omraade',  # chefområde
+                  'afdeling',  # nærmeste leders afdeling
+                  'gruppe',  # engagmentsafdeling
+                  'org_sti',
+                  'stilling',
+                  'engagements_type']
+
+    rows = []
+
+    for node in PreOrderIter(nodes['root']):
+        ou = mh.read_ou(node.name)
+        # Do not add "Afdelings-niveau"
+        if ou['org_unit_level']['name'] != 'Afdelings-niveau':
+            employees = mh.read_organisation_people(node.name, 'engagement', False)
+            manager = find_org_manager(mh, node)
+
+            org_path = f"{ou['location']}/{ou['name']}"
+
+            for uuid, employee in employees.items():
+                row = {}
+                address = mh.read_user_address(uuid, username=True, cpr=True)
+
+                """
+                Create generic mapping function from employee data to fieldnames
+                """
+
+
+def export_to_orgviewer(mh, all_nodes):
+
+    # Root must be Holstebro Kommune for org viewer
+    roots = SETTINGS['exports.holstebro.roots']
+    org_root_name = SETTINGS['municipality.name']
+    org_root_uuid = SETTINGS['municipality.uuid']
+
+    data_fields = {'name': 'Location', 'value': '', 'type': 'text'}
+    people = []
+    assignments = []
+
+    # Manually add root org unit "Holstebro Kommune" from settings and
+    # add manager to assignments and people
+    data = {'api_version': '1.0',
+            'chart': {
+                'id': org_root_uuid,
+                'name': org_root_name,
+                'description': '',
+                'parent_id': '',
+                'manager_id': SETTINGS['exports.holstebro.root_manager'],
+                'staff_department': 'N',
+                'dataFields': [data_fields],
+                'children': [],
+                "showChildren": True
+            },
+            'people': people,
+            'assignments': assignments
+            }
+
+    # this is where we start adding children
+    # Create dict with {'parent_uuid': ref to children[]}
+    children_ref = {}
+    children_ref.update({org_root_uuid: data['chart']['children']})
+
+    logger.info(f"Exporting root unit: {org_root_name} to Org. viewer")
+
+    # is this a partial exports
+    export_all = True if len(roots[org_root_uuid]) == 0 else False
```

Comment:
Dette burde være ækvivalent:
```suggestion
    export_all = not roots[org_root_uuid]
```